aws/data_source_aws_subnet_test.go

```diff
@@ -48,6 +49,8 @@ func TestAccDataSourceAwsSubnet_basic(t *testing.T) {
 						ds1ResourceName, "tags.Name", tag),
 					resource.TestMatchResourceAttr(
 						ds1ResourceName, "arn", arnregex),
+					resource.TestMatchResourceAttr(
```

Comment:
@drexler Thanks for this.
Could you please use [`testAccMatchResourceAttrRegionalARN`](https://github.com/terraform-providers/terraform-provider-aws/blob/0286dba6e6f55c52863bdfa6662282a250a32a74/aws/provider_test.go#L126) or equivalent function here over using an explicit regex (and also change the subnet ARN checks as well)?

Reply:
Thanks for the review. Since this ARN might be empty for subnets not linked to AWS outputs, what do you think of the approach of using a simple check for the output_arn via `TestCheckResourceAttr` and then once we have support for Outposts in the aws_subnet resource, adding tests here to utilize [ testAccMatchResourceAttrRegionalARN](https://github.com/terraform-providers/terraform-provider-aws/blob/0286dba6e6f55c52863bdfa6662282a250a32a74/aws/provider_test.go#L126) for cases where an ARN is expected?